pentagon/cli.py

```diff
@@ -65,18 +74,111 @@ def start_project(name, **kwargs):
+@click.option('--log-level', default="INFO", help="Log Level DEBUG,INFO,WARN,ERROR")
+@click.argument('additional-args', nargs=-1, default=None)
+def add(ctx, component_path, additional_args, **kwargs):
+    _run('add', component_path, additional_args, kwargs)
+
+
+@click.command()
+@click.pass_context
+@click.argument('component_path')
+@click.option('--data', '-D', multiple=True, help='Individual Key=Value pairs used by the component')
+@click.option('--file', '-f', help='File to read Key=Value pair from (yaml or json are supported)')
+@click.option('--out', '-o', default='./', help="Path to output module result, if any")
+@click.option('--log-level', default="INFO", help="Log Level DEBUG,INFO,WARN,ERROR")
+@click.argument('additional-args', nargs=-1, default=None)
+def get(ctx, component_path, additional_args, **kwargs):
+    _run('get', component_path, additional_args, kwargs)
+
+
+def _run(action, component_path, additional_args, options):
+    logging.basicConfig(level=options.get('log_level'))
+    logging.debug("Importing module Pentagon{}".format(component_path))
+    logging.debug("with options: {}".format(options))
+    logging.debug("and additional arguments: {}".format(additional_args))
+
+    component_class = get_component_class(component_path)
+
+    # Process infile or data options
+    data = {}
     try:
-        pentagon.PentagonComponent(name, kwargs).install()
+        file = options.get('file', None)
+        if file is not None:
+            data = parse_infile(file)
+        data = parse_data(options.get('data', data))
+
     except Exception as e:
+        logging.error("Error parsing data from file or -D arguments")
+        logging.error(e)
+        logging.debug(traceback.print_exc(e))
+
+    try:
+        if action == 'get':
+            component_class(data, additional_args).get(options.get('out'))
+        if action == 'add':
+            component_class(data, additional_args).add(options.get('out'))
+    except Exception, e:
         logging.error(e)
         logging.debug(traceback.print_exc(e))
 
+
 # Making names more terminal friendly
-cli.add_command(install_component, "install-component")
 cli.add_command(start_project, "start-project")
+cli.add_command(add, "add")
+cli.add_command(get, "get")
+
+
+def get_component_class(component_path):   # Construct Class path from component input
+    component_path_list = component_path.split(".")
+    if len(component_path_list) > 1:
+        component_name = ".".join(component_path.split(".")[0:-1])
+        component_class_name = component_path.split(".")[-1].title()
+    else:
+        component_name = component_path
+        component_class_name = component_path.title()
+
+    # Find Class if it exists
+    component_class = locate("pentagon.component.{}.{}".format(component_name, component_class_name))
+    if component_class is None:
+        component_class = locate("pentagon{}.{}".format(component_name, component_class_name))
+
+    if component_class is None:
```

Comment:
I think an exception should be raised after logging in this block. Otherwise `None` could be returned and cause a failure in `_run` when a less helpful exception  of `TypeError: 'NoneType' object is not callable`

Reply:
I'll rework a little bit I don't think `get_component_class` should throw an exception, but I agree that a better error message is in order.